pkg/platform/kube/functionres/lazy.go

```diff
@@ -597,6 +614,22 @@ func (lc *lazyClient) createOrUpdateDeployment(functionLabels labels.Set,
 	return resource.(*apps_v1beta1.Deployment), err
 }
 
+func enrichDeploymentSpec(dest *apps_v1beta1.DeploymentSpec, source apps_v1beta1.DeploymentSpec) error {
+	encodedDeploymentSpec, err := json.Marshal(dest)
+	if err != nil {
+		return errors.Wrap(err, "Failed to marshal destination deployment spec")
+	}
+
+	err = json.Unmarshal(encodedDeploymentSpec, &source)
```

Comment:
This seems like an implicit way of merging using json unmarshelling properties. We have `mergo`. Why not use it?